src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilder.java

```diff
@@ -0,0 +1,422 @@
+        pointHandleShape.getAttributes().setSelectionBoundsOffset(SELECTION_OFFSET);
+        pointHandleShape.getAttributes().setSelectionStrokeOffset(SELECTION_OFFSET);
+        pointHandleShape.setFillBoundsForSelection(true);
+        pointHandleShape.setFillShapeForSelection(true);
+        pointHandleShape.setListening(true);
+        return pointHandleShape;
+    }
+
+    private int getPointNearTo(final Point2D location)
+    {
+        final Point2DArray linePoints = connector.getLine().getPoint2DArray();
+        for (int i = 0; i < linePoints.size(); i++)
+        {
+            final Point2D point = linePoints.get(i);
+            if (Geometry.distance(location.getX(), location.getY(), point.getX(), point.getY()) < R1)
+            {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private void destroyMousePointerCP()
+    {
+        if (null != mousePointerCP)
+        {
+            mousePointerCP.removeFromParent();
+            mousePointerCP = null;
+        }
+        closeControlPointBuildAnimation();
+        batch();
+    }
+
+    private void register(final HandlerRegistrationManager manager,
+                          final HandlerRegistration registration)
+    {
+        registrations.add(registration);
+        manager.register(registration);
+    }
+
+    private void cancelExitTimer()
+    {
+        if (null != exitTimer)
+        {
+            exitTimer.cancel();
+            exitTimer = null;
+        }
+    }
+
+    private void forceRunExitTimer()
+    {
+        if (null != exitTimer && !exitTimer.isRunning())
+        {
+            exitTimer.run();
+            exitTimer = null;
+        }
+    }
+
+    private void batch()
+    {
+        getLayer().batch();
+        getTransientPointLayer().batch();
+    }
+
+    private WiresConnectorControlImpl getControl()
+    {
+        return (WiresConnectorControlImpl) connector.getControl();
+    }
+
+    private Layer getTransientPointLayer()
+    {
+        return getOverLayer();
+    }
+
+    private Layer getLayer()
+    {
+        return connector.getGroup().getLayer().getLayer();
+    }
+
+    private Layer getOverLayer()
+    {
+        return getLayer().getOverLayer();
+    }
+
+    private static String figureOutControlPointBuilderColor(final Shape<?> mousePointerDecorator)
+    {
+        String color = null;
+        if (mousePointerDecorator.getStrokeAlpha() > 0
+            && !isWhite(mousePointerDecorator.getStrokeColor()))
+        {
+            color = mousePointerDecorator.getStrokeColor();
+        }
+        else if (mousePointerDecorator.getFillAlpha() > 0
+                 && !isWhite(mousePointerDecorator.getFillColor()))
+        {
+            color = mousePointerDecorator.getFillColor();
+        }
+        return null != color ? color : DEFAULT_CP_BUILDER_SHAPE_COLOR;
+    }
+
+    private static boolean isWhite(final String color)
```

Comment:
This method is always inverted in the code. Probably, we can use `isNotWhite` instead.

Reply:
fixed